cocos/particle/renderer/trail.ts

```diff
@@ -351,11 +351,14 @@ export default class TrailModule {
         this._particleSystem = ps;
         this.minParticleDistance = this._minParticleDistance;
         let burstCount = 0;
-        for (const b of this._particleSystem.bursts) {
-            burstCount += b.getMaxCount(this._particleSystem);
-        }
-        this._trailNum = Math.ceil(this._particleSystem.startLifetime.getMax() * this.lifeTime.getMax() * 60 * (this._particleSystem.rateOverTime.getMax() * this._particleSystem.duration + burstCount));
-        this._trailSegments = new Pool(() => new TrailSegment(10), Math.ceil(this._particleSystem.rateOverTime.getMax() * this._particleSystem.duration));
+        const psTime = ps.startLifetime.getMax();
+        const psRate = ps.rateOverTime.getMax();
+        const duration = ps.duration;
+        for (const b of ps.bursts) {
```

Comment:
Please do not use for of, use standard for or for in instead

Reply:
> Please do not use for of, use standard for or for in instead

This has been modified.